src/modules/consumer.js

```diff
@@ -61,31 +68,29 @@ class Consumer {
     if (typeof options === 'function') {
       callback = options;
       // default message options
-      options = { persistent: true, durable: true };
+      options = { persistent: true, durable: true, channel: {} };
     }
 
     // consumer gets a suffix if one is set on the configuration, to suffix all queues names
     // ex: service-something with suffix :ci becomes service-suffix:ci etc.
     const suffixedQueue = `${queue}${this._connection.config.consumerSuffix || ''}`;
 
     return this._connection
-      .get()
+      .getChannel(queue, options.channel || {})
       .then((channel) => {
-        this.channel = channel;
-
         // when channel is closed, we want to be sure we recreate the queue ASAP so we trigger a reconnect by recreating the consumer
-        this.channel.addListener('close', () => {
+        channel.addListener('close', () => {
```

Comment:
If the inner `Channels._get` functions fails with an unexpected error it doesn't reject, just resolves with the `channel` here as `undefined`.
That's how it was before your changes as well, but maybe you can address that.

Reply:
Changed `_get` method catch handler to return rejected promise.